File: linkedList.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node Node;
typedef struct LinkedList LinkedList;

struct Node {
    int value;
    Node *nextNode;
};

Node * newNode(int value) {
    Node * newNode = (Node *)malloc(sizeof(Node));
    newNode->value = value;
    newNode->nextNode = NULL;
    return newNode;
}

struct LinkedList {
    Node *head;
};

LinkedList * newLinkedList() {
    LinkedList * newLinkedList = (LinkedList *)malloc(sizeof(LinkedList));
    newLinkedList->head = NULL;
    return newLinkedList;
}

File: linkedList.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def push(self, value):
        node = Node(value=value)
        if self.head == None:
            self.head = node
        else:
            aux = self.head
            while aux.next != None:
                aux = aux.next
            aux.next = node

    def pop(self):
        aux = self.head
        if aux.next == None:
            self.head - None
        else:
            while aux.next.next != None:
                aux = aux.next
            aux.next = None

    def printList(self):
        aux = self.head
        while aux != None:
            print(aux.value)
            aux = aux.next



